frontend2/src/views/MyTeam.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useMemo, useState } from "react";
+import { PageTitle } from "../components/elements/BattlecodeStyle";
+import SectionCard from "../components/SectionCard";
+import Input from "../components/elements/Input";
+import TextArea from "../components/elements/TextArea";
+import { TeamStateEnum, useCurrentTeam } from "../contexts/CurrentTeamContext";
+import Button from "../components/elements/Button";
+import MemberList from "../components/team/MemberList";
+import DescriptiveCheckbox from "../components/elements/DescriptiveCheckbox";
+import JoinTeam from "../components/JoinTeam";
+
+const MyTeam: React.FC = () => {
+  const { team, teamState } = useCurrentTeam();
+  const [checked, setChecked] = useState<boolean>(false);
+  const membersList = useMemo(() => {
+    return (
+      <div className="flex flex-col gap-8">
+        {team !== undefined && <MemberList members={team?.members} />}
+        <Button className="self-start" label="Leave team" />
+      </div>
+    );
+  }, [team]);
+  if (teamState !== TeamStateEnum.IN_TEAM || team === undefined) {
+    return <JoinTeam />;
+  }
+  return (
+    <div className="p-10">
+      <PageTitle>Team Settings</PageTitle>
+      <div className="flex flex-col gap-8 xl:flex-row">
+        <div className="flex flex-1 flex-col gap-8 xl:max-w-4xl">
+          <SectionCard title="Profile" className="max-w-5xl">
+            <div className="flex flex-col md:flex-row md:gap-8">
+              <div className="flex flex-col items-center gap-6 p-4">
+                <img
+                  className="h-24 w-24 rounded-full bg-gray-400 md:h-48 md:w-48"
+                  src={team.profile?.avatar_url}
+                />
+                <div className="text-center text-xl font-semibold">
+                  {team.name}
+                </div>
+              </div>
+              <div className="flex flex-1 flex-col gap-4">
+                <Input
+                  disabled
+                  label="Join key"
+                  value={team.join_key ?? "Loading..."}
```

Comment:
Should we add a spinner here?

Reply:
b/c it's an input field I'm hesitant to add a spinner; we may want to consider making the whole page a spinner while the team is loading instead, but that can be added later